### PR TITLE
[netcore] Fix CI build script

### PR DESCRIPTION
**Pull request by EgorBo**
Also fix `make run-sample`

### CODE REVIEW

File: scripts/ci/pipeline-netcore-runtime.yml [hunk 1]
```diff
@@ -82,10 +82,16 @@ jobs:
 
   - bash: |
       ./autogen.sh --with-core=only
-      make -j 4
+      make -C mono -j4
```
Comment (marek-safar) [inline scripts/ci/pipeline-netcore-runtime.yml]: This is redundant, make builds everything

Comment (EgorBo) [reply scripts/ci/pipeline-netcore-runtime.yml]: @marek-safar I don't know why but `make` from the root folder sometimes hangs (happened for @steveisok too afair)

Comment (marek-safar) [reply scripts/ci/pipeline-netcore-runtime.yml]: does it still happen? I thought I fixed by removing that recursive target from netcore

Comment (EgorBo) [reply scripts/ci/pipeline-netcore-runtime.yml]: happened for me 15 minutes ago :(

Comment (EgorBo) [reply scripts/ci/pipeline-netcore-runtime.yml]: hm.. `build.sh` used to run xunit tests via ./run-step script (which could handle timeouts) now the macOS lane is dead cause `make -C netcore run-tests-corefx` can't do it.

**Comment by directhex** [comment]
that's new. hello bot. where'd you come from? :grimacing:

**Comment by directhex** [comment]
oh, the slash-azp. interesting

File: scripts/ci/pipeline-netcore-runtime.yml (hunk 1 above)
Comment (steveisok) [reply scripts/ci/pipeline-netcore-runtime.yml]: make -j <any number> seems to hang all the time for me.  Just plain make does not.

Comment (marek-safar) [reply scripts/ci/pipeline-netcore-runtime.yml]: could you debug that, I tried 10 times in a row with clean configure with no hang with tip of master

**Pull request merged**